lib/backend/block_affinity.go

```diff
@@ -44,7 +44,19 @@ func (key BlockAffinityKey) asEtcdKey() (string, error) {
 }
 
 func (key BlockAffinityKey) asEtcdDeleteKey() (string, error) {
-	return key.asEtcdKey()
+	// Must specify at least a host to remove.
+	if key.Host == "" {
+		return "", common.ErrorInsufficientIdentifiers{}
+	}
+	k := "/calico/ipam/v2/host/" + key.Host
```

Comment:
I think when deleting a resource we need to be fully explicit about which resource we delete - it's too easy to make mistakes otherwise and get into all sorts of trouble.